pandas/core/indexes/base.py

```diff
@@ -1086,6 +1087,11 @@ def astype(self, dtype, copy: bool = True):
 
         values = self._data
         if isinstance(values, ExtensionArray):
+            if isinstance(dtype, np.dtype) and dtype.kind == "M" and is_unitless(dtype):
```

Comment:
Is this for users who may have created a custom EA w/ dtype with these conditions? Would it need a deprecation as well?

Reply:
this is to keep the 1.4.x behavior for Index

Reply:
Got it, so we have a test that covers this case?

Reply:
yes